docs/arch/adr-5.md

```diff
@@ -0,0 +1,30 @@
+# ADR 5: Implementing workflows in the new classifier
+
+July 25, 2018
+
+## Context
+
+The current classifier was based on a few assumptions, that, while accurate at the time, became outdated as additional functionality was added:
+
+1. That a workflow step consisted of a single annotation action. However, project builders wanted to e.g. annotate via drawing _and_ selecting from a list of options in a single step, which gave rise to the combo task.
+1. That we would always want to show a summary. When the `hide classification summaries` tool became public, more projects actually wanted to use it than we expected.
+1. That we wouldn't need to show any information to a volunteer until the end. That changed firstly with the MS interventions experiment, and later feedback, which is shown once a workflow task is completed.
+
+## Decision
+
+We implement the classification process like this:
+
+The classification of a subject will consist of a series of __steps__. A single step consists of a __task hook__, and a __notification hook__. A task hook may consist of a __single workflow task__, or an __array of workflow tasks__. A notification could be an __intervention__, __feedback__, a __Sugar notification__, or some other information conveyed to the volunteer.
```

Comment:
I'd get rid of single workflow task here, and just have an array of workflow tasks (where the array might only have one task for most workflows.) I don't think there would be a performance problem, and it would avoid code that has to switch based on whether or not the next step is a task or an array.